src/mod_mam_utils.erl

```diff
@@ -969,23 +969,58 @@ maybe_previous_id(X) ->
     X - 1.
 
 
--spec is_last_page(PageSize, TotalCount, Offset, MessageRows) -> boolean() when
+%% @doc Returns true, if the current page is the final one in the result set.
+%% If there are more pages with messages, than this function returns false.
+%%
+%% PageSize - maximum number of messages extracted in one lookup.
+%% TotalCount - total number of messages in the Result Set.
+%% Result Set - is subset of all messages in user's archive,
+%% in a specified time period.
+%% MessageRows - stuff we are about to send to the user.
+%% Params - lookup parameters, coming from mam_iq module.
+%%
+%% TotalCount and Offset can be undefined, in case we use IsSimple=true.
+%% IsSimple=true tells the server not to do heavy `SELECT COUNT(*)' queries.
+%%
+%% TODO It is possible to set complete flag WITH IsSimple=true,
+%%      if we select one extra message from archive, but don't send it to the client.
+%%      It's the most efficient way to query archive, if the client side does
+%%      not care about the total number of messages and if it's stateless
+%%      (i.e. web interface).
+-spec is_complete_result_page(PageSize, TotalCount, Offset, MessageRows, Params) ->
```

Comment:
I think we can assume that `PageSize` is a part of `Params`?

Reply:
Unless it's get min(PageSizeFromClient, MaxPageSizeInOpts).
Need to check.